MC-11912:API-document-for-PVC (#194)
cloudops/cloudmc-api-docs: source/includes/kubernetes_extension/_k8_persistentvolumeclaims.md

               "release":"cmc-staging"
            },
            "name":"cmc-staging-mysql",
            "namespace":"cmc-stg",
            "resourceVersion":"111495311",
            "selfLink":"/api/v1/namespaces/cmc-stg/persistentvolumeclaims/cmc-staging-mysql",
            "uid":"ba115a65-e3c5-4e37-8548-e37ec395e79f"
         },
         "spec":{
            "accessModes":[
               "ReadWriteOnce"
            ],
            "resources":{
               "requests":{
                  "storage":"8Gi"
               }
            },
            "storageClassName":"rook-ceph-block",
            "volumeMode":"Filesystem",
            "volumeName":"pvc-ba115a65-e3c5-4e37-8548-e37ec395e79f"
         },
         "status":{
            "accessModes":[
               "ReadWriteOnce"
            ],
            "capacity":{
               "storage":"8Gi"
            },
            "phase":"Bound"
         }
      }
   ],
   "metadata":{
      "recordCount":1
   }
}
```

<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumeclaims?cluster_id=:cluster_id</code>

> Retrieve a list of all persistent volumes in a given [environment](#administration-environments).

Reviewer: ```suggestion
Retrieve a list of all persistent volume claims in a given [environment](#administration-environments).
```
?

| `id` <br/>_string_ | The id of the persistent volume claim. This is the name of the resource. |
| `metadata` <br/>_object_ | The metadata of the persistent volume claim.|
| `spec` <br/>_object_ | The specification of the persistent volume claim. |
| `spec.accessModes` <br/>_string_ | The volume can be mounted on a host in any way supported by the resource provider and will give the provider access to different capabilities. Value is one of `ReadWriteOnce` (by a single node), `ReadOnlyMany` (by many nodes) or `ReadWriteMany` (by many nodes). |
| `spec.resources.requests.storage` <br/>_string_ | Requested storage capacity of the persistent volume. |
| `spec.storageClassName` <br/>_string_ | Storage class associated to the volume. |
| `spec.capacity.volumeMode` <br/>_string_ | If set to `Filesystem` (default value), the volume is mounted into Pods into a directory. If set to `Block`, then the volume is used as a raw block device. |
| `status.phase` <br/>_string_ | Volume is in one of the following phases: `Available`, `Bound`, `Released` or `Failed`. |

<!-------------------- GET A persistent volume -------------------->

> #### Get a persistent volume

Reviewer: ```suggestion
#### Get a persistent volume claim
```

      "finalizers": [
        "kubernetes.io/pvc-protection"
      ],
      "labels": {
        "app": "cmc-staging-mysql",
        "chart": "mysql-0.4.5",
        "heritage": "Helm",
        "release": "cmc-staging"
      },
      "name": "cmc-staging-mysql",
      "namespace": "cmc-stg",
      "resourceVersion": "111495311",
      "selfLink": "/api/v1/namespaces/cmc-stg/persistentvolumeclaims/cmc-staging-mysql",
      "uid": "ba115a65-e3c5-4e37-8548-e37ec395e79f"
    },
    "spec": {
      "accessModes": [
        "ReadWriteOnce"
      ],
      "resources": {
        "requests": {
          "storage": "8Gi"
        }
      },
      "storageClassName": "rook-ceph-block",
      "volumeMode": "Filesystem",
      "volumeName": "pvc-ba115a65-e3c5-4e37-8548-e37ec395e79f"
    },
    "status": {
      "accessModes": [
        "ReadWriteOnce"
      ],
      "capacity": {
        "storage": "8Gi"
      },
      "phase": "Bound"
    }
  }
}
```

<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumeclaims/:id?cluster_id=:cluster_id</code>

> Retrieve a persistent volume and all its info in a given [environment](#administration-environments).

Reviewer: ```suggestion
Retrieve a persistent volume claim and all its info in a given [environment](#administration-environments).
```

> Retrieve a persistent volume claims and all its info in a given [environment](#administration-environments).

Reviewer: ```suggestion
Retrieve a persistent volume claim and all its info in a given [environment](#administration-environments).
```
my bad this one should be singular since its get one instead of list
